README.md

```diff
@@ -118,12 +128,22 @@ wget https://sparse.tamu.edu/RB/Nasa/nasa2146.tar.gz
+upcxx-run -n 4 -- ./run_sympack2D -in nasa2146/nasa2146.rb -ordering MMD -nrhs 1
 ```
 For larger problems, like [audikw_1](https://www.cise.ufl.edu/research/sparse/RB/GHS_psdef/audikw_1.tar.gz), it is strongly advised to use more efficient orderings (such as **METIS**).
-Note that to run **symPACK**, `upcxx-run` is used in the example above, but on some platforms, such as NERSC Cori,
+Note that to run **symPACK**, `upcxx-run` is used in the example above, but on some platforms, such as NERSC Perlmutter,
 other launchers may be used to both spawn **MPI** and **UPC++**, such as `srun` if the system is using SLURM.
-Moreover, for larger problems, the `-shared-heap XX` argument to `upcxx-run` may be needed (Please refer to **UPC++** documentation).
+Moreover, for larger problems, the `-shared-heap XX` argument to `upcxx-run` may be needed (Please refer to [**UPC++** documentation](https://upcxx.lbl.gov/site)).
+
+# GPU Mode Options
+---------------------------
+**SymPACK** provides several optional command line options that allow the user to configure certain aspects of the GPU mode. They are summarized here:
+
+- `-gpu_mem <SIZE>` controls the amount of memory (in bytes) allocated on the GPU upon calling the `symPACK_cuda_setup()` method. If this option is not specified, **symPACK** will attempt partition a device's memory equally among each process bound to the device. For job configurations that have more than one GPU bound to each node, this default behavior may under-estimate the amount of memory that can be allocated on each device. Therefore, for these sorts of job configurations, it is strongly recommended to use this command line option. 
+- `-gpu_blk <SIZE>` controls the size threshold (in bytes) that a factorized diagonal block must exceed for it to be automatically copied to the GPU bound to a remote process when being sent to said remote process. This allows for an intermediate copy to host memory to be bypassed, which can reduce communication overhead for certain problems.
+- `{trsm, gemm, potrf, syrk}_limit <SIZE>` controls how many nonzero entries a block must have for the given matrix operation (TRSM, GEMM, POTRF, or SYRK) involving said block to be offloaded to the GPU. The GPU will generally be most beneficial for operations involving large blocks with many nonzeros, but the exact size thresholds for when each operation starts to run faster on the GPU will differ between devices.
+- `-gpu_solve` determines if the GPU will be used for **symPACK's** triangular solve routine.
+- `-fallback {cpu, terminate}` determines what **symPACK** will do in the event that an attempt to allocate memory on a device for use in a matrix computation fails due to an insufficient amount of free memory. Setting this option to `cpu` will make it so **symPACK** will simply perform the computation on the CPU instead of the GPU. Setting this option to `terminate` will cause **symPACK** to throw an error and cease execution when a device allocation fails. 
```

Comment:
Would like us to say more about the default behavior for command-line arguments.

I suggest at least:

```suggestion
- `-gpu_solve` determines if the GPU will be used for **symPACK's** triangular solve routine. Defaults to off (use the CPU for the triangular solve).
- `-fallback {cpu, terminate}` determines what **symPACK** will do in the event that an attempt to allocate memory on a device for use in a matrix computation fails due to an insufficient amount of free memory. Setting this option to `cpu` will make it so **symPACK** will simply perform the computation on the CPU instead of the GPU. Setting this option to `terminate` will cause **symPACK** to throw an error and cease execution when a device allocation fails. Defaults to `terminate`
```


I'm less sure we should encode the `-gpu_mem` and `-gpu_blk` numeric defaults in this document. Ideally we'd have a `-v` option to enable verbosity that reports the active values of all these voodoo tuning parameters. 


Thoughts?

Reply:
Yeah the default values of each of these definitely need to be specified in the README. It would be easy and helpful to add a verbosity option, so I will do so.

Reply:
verbosity option added in commit [f6d8093] (https://github.com/symPACK/symPACK/pull/12/commits/f6d809341bcba5312b9f1752663877df14e58e6e)